### ISSUE TITLE
Storybook: Table component under Visualizations throws error

### ISSUE THREAD

**Issue by JoaoSilvaGrafana** [grafana/grafana#53684]
**What happened**:

If you go to any story of Visualizations > Table locally on storybook, an error is thrown:
![imagen](https://user-images.githubusercontent.com/100691367/184403356-838b81ff-a963-4ba8-bca1-e6278266b9cc.png)

**What you expected to happen**:

Storybook to display the component

**How to reproduce it (as minimally and precisely as possible)**:

Run storybook locally and go to http://localhost:9001/?path=/story/visualizations-table--bar-gauge-cell

**Environment**:
- Grafana version: 
- Data source type & version: -
- OS Grafana is installed on: -
- User OS & Browser: -
- Grafana plugins: -
- Others: -

**Comment by matyax** [comment]
This seems to be intermitently failing for some reason. I re-tested this a few days ago, didn't fail, now ran storybook again and reproduced the error.

**Comment by JoaoSilvaGrafana** [comment]
I've just tested locally and it seems broken for various of the visualizations components actually! Assuming we added something to visualizations and did not update these components. I will investigate

**Comment by JoaoSilvaGrafana** [comment]
This is broken as of this PR: https://github.com/grafana/grafana/pull/51517 which is a pretty big change so hard to pinpoint what exactly is causing the issue... I will put this into our backlog of things to solve.

What seems to be happening is that the `fieldColorModeRegistry` in [fieldColor.ts](https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/field/fieldColor.ts#L24) is no longer getting initialized.

**Comment by joshhunt** [comment]
Investigated this in mob today:

The `init` property in Registry class is not being properly set with the shorthand `constructor(private init?: () => T[]) {}` - this should set the `init` argument to the property `init` on the class. At the moment, In storybook, it's being transpiled to:

```
var Registry = /*#__PURE__*/
function() {
    function Registry(init) {
        var _this = this;

        _classCallCheck(this, Registry);
        // ...
```

When we remove the peer dependencies from `grafana-data`, Storybook transpiles the Registry class like:

```
var Registry = /*#__PURE__*/
function() {
    function Registry(init) {
        var _this = this;

        _classCallCheck(this, Registry);
        this.init = init; // <---- init is being set!!
        // ...
```

When we `yarn build` `grafana-data`, the file is transpiled (with esbuild) correctly, with `init` being set, so the problem is unique to Storybook.

If we explicitly set init in the `Registry` constructor (e.g. `constructor(init) => this.init = init`) it works, but we don't want to change our real code just for Storybook, and we don't know where else we rely on this pattern that would need updating.

It's unclear why Storybook(?) is transpiling the code differently depending on the peer deps of `grafana-data` - from logging the babel and storybook config don't seem any different.

Issue seems related to https://github.com/storybookjs/storybook/issues/13834, though this does not mention anything about peer deps. We briefly tried a few of the fixes in here and they didn't work (or caused other issues), but would still be good to get to the bottom of why peer deps changes this.

@jackw Does any of this sound familiar to you?

**Comment by jackw** [comment]
@joshhunt @JoaoSilvaGrafana it's not familiar to me however I have spent a bit of time recently updating all our storybook deps and the storybook config to remove as much of the customisation as possible.

Try cloning #49793 and you should see the Table stories locally...

<img width="1679" alt="Screenshot 2022-09-27 at 14 24 24" src="https://user-images.githubusercontent.com/73201/192525844-623378c5-9029-4842-8679-98fec31d1aff.png">

That PR is slightly blocked by the storybook-dark-mode addon that doesn't play nicely with latest storybook and yarn pnp. I'm working on a fix for that in the coming days.

**Comment by JoaoSilvaGrafana** [comment]
Thanks for the reply @jackw I have added a quick fix for the 9.2 version so these aren't broken until 9.3 👍
